website/docs/server/data-loader/data-loader.md

```diff
@@ -97,8 +97,97 @@ class User(val id: ID) {
 }
 ```
 
-:::info
-Given that `graphql-java` relies on `CompletableFuture`s for scheduling and asynchronous execution of `DataLoader` calls,
-currently we don't provide any native support for `DataLoader` pattern using coroutines. Instead, return
+## DataLoaders and Coroutines
+
+`graphql-java` relies on `CompletableFuture` for scheduling and execute asynchronous field resolvers (aka data fetchers),
+`graphql-java` deliberately aims for near zero dependencies which means no Kotlin / WebFlux / RxJava.
+
+Similar to `asynchronous` field resolvers (aka data fetcher), `DataLoader` pattern implementation in `graphql-java` works with `CompletableFuture`,
+and because of the listed `graphql-java` constrains we don't provide any native support for `DataLoader` pattern using suspendable functions. Instead, return
```

Comment:
IMHO previous version was clearer - I don't think adding the section about the data fetchers is needed and it brings confusion, i.e. yes `graphql-java` does use CF for data fetchers but we provide interop with coroutines but also `graphql-java` uses CF for data loaders but we don't provide support for it?

Adding details on why it works on data fetchers but not on data loaders might be too much so I'd just keep the previous comment.